File: template.py
# TODO URL

from collections import *
from functools import *
import itertools
import re
import sys

def solve(data):
    for line in data.splitlines():
        print(line)

########################################################################
#                             SETUP STUFF                              #
########################################################################

sample_input = r"""
""".strip()

actual_input = r"""
""".strip()

if len(sys.argv) > 1:
    with open(sys.argv[1], 'r') as f:
        file_input = f.read().strip()
    solve(file_input)
else:
    print('=== SAMPLE ===')
    solve(sample_input)
    print('=== ACTUAL ===')
    solve(actual_input)

File: template.cc
// TODO URL

#include <bits/stdc++.h>

#define all(x) begin(x), end(x)
#define sz(x) ((int)(x).size())
using ll = long long;
using namespace std;

void solve(istream& ins)
{
	assert(ins.good());
}

////////////////////////////////////////////////////////////////////////
//                            SETUP STUFF                             //
////////////////////////////////////////////////////////////////////////

static const string sample_input =
R"""( )""";

static const string actual_input =
R"""( )""";

int main(int argc, const char *argv[])
{
	if (argc > 1) {
		ifstream ifs {argv[1]};
		solve(ifs);
	} else {
		cout << "=== SAMPLE ===\n";
		istringstream iss {sample_input};
		solve(iss);
		cout << "=== ACTUAL ===\n";
		iss = istringstream {actual_input};
		solve(iss);
	}
	return 0;
}
